src/limiter/maximum_principle_limiter.cpp

```diff
@@ -207,17 +238,11 @@ void MaximumPrincipleLimiter<dim, nstate, real>::limit(
         // Apply limiter on solution values at quadrature points
         for (unsigned int istate = 0; istate < nstate; ++istate) {
             for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
-                soln_at_q[istate][iquad] = theta[istate] * (soln_at_q[istate][iquad] - soln_cell_avg[istate])
+                soln_coeff[istate][iquad] = theta[istate] * (soln_coeff[istate][iquad] - soln_cell_avg[istate])
```

Comment:
This is just a placeholder that we should return and show mathematically whether updating the solution coefficients ensures we can preserve the maximum principle instead of the solution at the quadrature nodes.